src/core/diamond.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+  Copyright (C) 2010-2018 The ESPResSo project
+  Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010
+    Max-Planck-Institute for Polymer Research, Theory Group
+
+  This file is part of ESPResSo.
+
+  ESPResSo is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  ESPResSo is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** \file
+    This file contains everything needed to create a start-up configuration
+    of (partially charged) diamond structure polymer chains with counterions
+    and salt molecules, assigning velocities to the particles and
+    cross-linking the polymers if necessary.
+
+    The corresponding header file is diamond.hpp.
+*/
+
+#include <cmath>
+#include <cstddef>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+
+#include "PartCfg.hpp"
+#include "bonded_interactions/bonded_interaction_data.hpp"
+#include "communication.hpp"
+#include "constraints.hpp"
+#include "constraints/ShapeBasedConstraint.hpp"
+#include "debug.hpp"
+#include "diamond.hpp"
+#include "global.hpp"
+#include "integrate.hpp"
+#include "random.hpp"
+#include "utils.hpp"
+
+#include "utils/vec_rotate.hpp"
+using Utils::vec_rotate;
```

Comment:
```vec_rotate``` is not used

Reply:
Point of this PR is not re-implementing the diamond code, which was part of the old polymer.cpp file. I just put it into a different file.